Tasks/Database/database_rewiew.cpp

```diff
@@ -0,0 +1,148 @@
+#include <iostream>
+#include <iomanip>
+#include <vector>
+#include <string>
+#include <map>
+#include <set>
+
+class Date
+{
+public:
+    Date(int year, int month, int day): m_year(year), m_month(month), m_day(day) {};
+    Date(std::istream& stream)
+    {
+        int stops[2] = {0, 0};
+        std::string string;
+        stream >> string;
+        for (int i = 1; i < string.size(); i++)
+        {
+            if (string[i] == '-' && isdigit(string[i-1]))
+            {
+                if (stops[0] == 0)
+                    stops[0] = i;
+                else
+                    stops[1] = i;
+            }
+        }
+        int year; int month; int day;
+        try {
+            year = std::stoi(string.substr(0, stops[0]));
+            month = std::stoi(string.substr(stops[0] + 1, stops[1] - stops[0] - 1));
+            day = std::stoi(string.substr(stops[1] + 1));
+        }
+        catch(std::invalid_argument& error)
```

Comment:
```suggestion
        catch(const std::invalid_argument& error)
```
А вообще ловить исключения, нужно только в 1 месте -- функции мейн. Здесь ты делаешь очень плохо по семантике